lib/auth-middleware/auth_middleware/__init__.py

```diff
@@ -309,3 +311,28 @@ def use_security_token_auth(func):
 #         abort(401)
 #     request.user_token_data = id_info
 #     return
+
+def verify_githubauth(token_str):
```

Comment:
After our discussion, I figured that using `x-tokenInfoUrl: https://api.github.com/user` in security scheme will let GitHub do the token validation directly. Please run some tests with valid and possibly invalid tokens to see the error messages and it might be possible to remove this method completely.